server/channels/web/context.go

```diff
@@ -29,6 +29,14 @@ type Context struct {
 
 // LogAuditRec logs an audit record using default LevelAPI.
 func (c *Context) LogAuditRec(rec *audit.Record) {
+	// finish populating the context data, in case the session wasn't available during MakeAuditRecord
```

Comment:
Just so I understand, when is this expected to happen?

Reply:
@enzowritescode It happens whenever an audit log is persisted, which we do in a defer after creating the audit record, e.g.:

``` go
	auditRec := c.MakeAuditRecord("login", audit.Fail)
	defer c.LogAuditRec(auditRec)
```

So we're filling in info that the original caller was fine with adding when they made the audit record (but the info wasn't available then).

Hope that helps?

Reply:
(Followed up via DM)